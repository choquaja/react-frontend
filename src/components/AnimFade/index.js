import React from 'react';
import PropTypes from 'prop-types';
import Transition from 'react-motion-ui-pack';

const enterState = { opacity: 1 };
const leaveState = { opacity: 0 };

export default function AnimFade(props) {
  return <Transition component="div" enter={enterState} leave={leaveState} {...props} />;
}

AnimFade.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};
