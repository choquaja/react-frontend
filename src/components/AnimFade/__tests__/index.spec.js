import React from 'react';
import { shallow } from 'enzyme';
import AnimFade from '../';

describe('AnimFade', () => {
  it('renders without crashing', () => {
    shallow(<AnimFade><div>Hello</div></AnimFade>);
  });

  it('renders with the right props', () => {
    const props = {
      children: <div>Hello World </div>,
    };
    const component = shallow(<AnimFade{...props} />);
    expect(component).toMatchSnapshot();
  });
});
