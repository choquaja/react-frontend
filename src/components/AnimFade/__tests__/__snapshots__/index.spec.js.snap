// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimFade renders with the right props 1`] = `
<Transition
  component="div"
  enter={
    Object {
      "opacity": 1,
    }
  }
  leave={
    Object {
      "opacity": 0,
    }
  }
  onEnter={[Function]}
  onLeave={[Function]}
  runOnMount={true}
>
  <div>
    Hello World 
  </div>
</Transition>
`;
